source/Game.cpp

```diff
@@ -0,0 +1,199 @@
+#include"Game.hpp"
+
+#include<sstream>
+#include<iostream>
+#include<cctype>
+#include<string>
+#include"Error.hpp"
+
+namespace dungeon {
+
+Game::Game(): isEnd(false) {
+  initialize();
+}
+
+void Game::welcome() {
+  ios.clear();
+  ios.print("hello, world");
+  ios.pause();
+}
+
+void Game::selectClass() {
+  ios.clear();
+  ios.print("Select your Class");
+  ios.print("A. Saber");
+  ios.print("B. Archer");
+  ios.print("C. Lancer");
+  // ios.print("D. Rider");
+  // ios.print("E. Caster");
+  // ios.print("F. Assassin");
+  ios.print("G. Berserker");
+  switch (ios.inputChar()) {
+  case 'A':
+    mainChar.type = Class::saber; break;
```

Comment:
As mentioned before, since there are specializations of each character, you should use inheritance, and you can construct the main character object here.